Fairy v0.6

LeeGodSRC opened the pull request:
There are a LOT of changes here, I will write them all very soon.

LeeGodSRC committed:
Initial change for 0.6

LeeGodSRC committed:
Merge remote-tracking branch 'upstream/v5' into v6

LeeGodSRC committed:
More works on v6

LeeGodSRC committed:
Moved protocol and packets to protocol module

LeeGodSRC committed:
Fixed a little bit

LeeGodSRC committed:
Removed independent versioning, all 0.6.0b1 now

LeeGodSRC committed:
Merge remote-tracking branch 'upstream/v5' into v6

# Conflicts:
#	io.fairyproject.modules/platform.bukkit/bukkit-visual/build.gradle
#	io.fairyproject.modules/platform.bukkit/bukkit-visual/src/main/java/io/fairyproject/bukkit/visual/VisualBlockService.java
#	io.fairyproject.platforms/bukkit-platform/build.gradle
#	io.fairyproject.platforms/mc-platform/build.gradle

LeeGodSRC committed:
Working now!

LeeGodSRC committed:
Java 16+ compatibility

LeeGodSRC committed:
Use shadowJar instead of our own relocator

LeeGodSRC committed:
Cache MavenUtil, speed up the searching process quite a bit

LeeGodSRC committed:
Cache MavenUtil, speed up the searching process quite a bit

LeeGodSRC committed:
Added hot loading library support again but better

LeeGodSRC committed:
Fixed some structures

LeeGodSRC committed:
Move NBTEditor to independent module and interface to bridge for unit test

LeeGodSRC committed:
Revamped unit testing system

LeeGodSRC committed:
Cleanup

LeeGodSRC committed:
Don't load and save config file in unit test

LeeGodSRC committed:
Fixed a bug where ImanityItem is not submitted properly

LeeGodSRC committed:
Fixed gradle plugin

LeeGodSRC committed:
improved test ability for MCProtocol and timing

LeeGodSRC committed:
Added PojoMapper and InMemoryRepository

LeeGodSRC committed:
Added ConsumerSubscriber.java and NameTagUpdateEvent.java

LeeGodSRC committed:
Fixed JUnit BEFORE_ALL statement

LeeGodSRC committed:
Temp reverted MockBukkitContext.java

LeeGodSRC committed:
Beginning of ContainerNode and EventNode

LeeGodSRC committed:
Updated PacketEvents, fixed shutdown error

LeeGodSRC committed:
Improved logging

Use TinyLog for app platform
Added Log.java interface to warp different logging systems

LeeGodSRC committed:
Removed ASM library from bukkit-platform

LeeGodSRC committed:
Removed AspectJ and any features related to it

LeeGodSRC committed:
Fixed some errors

LeeGodSRC committed:
Fixed NameTagService.java

LeeGodSRC committed:
Added name tag visibility support to name tag service

LeeGodSRC committed:
Merge branch 'v6' into revamp/container

# Conflicts:
#	io.fairyproject.modules/module.storage/bukkit-storage/src/main/java/io/fairyproject/bukkit/storage/ThreadedPlayerStorage.java
#	io.fairyproject.platforms/core-platform/src/main/java/io/fairyproject/container/ComponentRegistry.java
#	io.fairyproject.platforms/core-platform/src/main/java/io/fairyproject/container/ContainerContext.java
#	io.fairyproject.platforms/core-platform/src/main/java/io/fairyproject/container/ContainerPluginListener.java
#	io.fairyproject.platforms/core-platform/src/main/java/io/fairyproject/container/controller/AutowiredContainerController.java
#	io.fairyproject.platforms/core-platform/src/main/java/io/fairyproject/container/scanner/BaseClassPathScanner.java
#	io.fairyproject.platforms/core-platform/src/main/java/io/fairyproject/container/scanner/DefaultClassPathScanner.java
#	io.fairyproject.platforms/core-platform/src/main/java/io/fairyproject/event/EventBus.java

LeeGodSRC committed:
More changes

LeeGodSRC committed:
Fixed some bukkit module issues

LeeGodSRC committed:
0.6.1 - container rework

LeeGodSRC committed:
Fixed bugs in container rework

LeeGodSRC committed:
Added mc.map module

LeeGodSRC committed:
Added exception printing to BasePluginHolder.java

LeeGodSRC committed:
Some changes

LeeGodSRC committed:
Code clean up

LeeGodSRC committed:
Use new repository url

LeeGodSRC committed:
Merge branch 'revamp/container' into v6

# Conflicts:
#	io.fairyproject.platforms/bukkit-platform/src/main/java/io/fairyproject/bukkit/mc/BukkitMCInitializer.java

LeeGodSRC committed:
Issue clean up

LeeGodSRC committed:
Clean up code style issues

LeeGodSRC committed:
Clean up code style issues x2

LeeGodSRC committed:
Clean up code style issues x3

LeeGodSRC committed:
Dependencies clean up

LeeGodSRC committed:
Update deploy-snapshot.yml

LeeGodSRC committed:
Update README.md

LeeGodSRC committed:
Bumped packetevents version

LeeGodSRC committed:
Merge remote-tracking branch 'origin/v6' into v6

terminalsin commented on buildSrc/build.gradle:
```diff
@@ -20,18 +20,18 @@ gradlePlugin {
-    implementation 'org.ow2.asm:asm-commons:9.2'
+    implementation "com.google.guava:guava:31.0.1-jre"
+    implementation "org.apache.commons:commons-lang3:3.12.0"
+    implementation "org.ow2.asm:asm:9.2"
```
ASM should be upgraded to 9.3 to support JDK 17+

terminalsin replied:
I would recommend having a gradle.settings file with all the versions as opposed to having them all over the project

LeeGodSRC replied:
done

terminalsin commented on gradle-plugin/build.gradle:
```diff
@@ -40,32 +40,29 @@ pluginBundle {
+    implementation libs.gson
+    implementation libs.guava
+    implementation libs.common.lang3
     implementation 'org.ow2.asm:asm:9.2'
```
Same goes here. ASM needs a version upgrade

LeeGodSRC replied:
done

terminalsin commented on gradle-plugin/src/main/java/io/fairyproject/gradle/util/MavenUtil.java:
```diff
@@ -35,55 +99,124 @@ public String getLatest(String module) throws IOException {
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setDoInput(true);
+            connection.setRequestMethod("GET");
+            connection.addRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:56.0) Gecko/20100101 Firefox/56.0");
```
User agent should be a custom one if you want to do analytics tracking

LeeGodSRC replied:
done

terminalsin reviewed (commented):
Still a WIP review since I didn't have time to go over everything; Here's my current comments:

- Love the new outline and features. Seems great ALTHOUGH I am really not a fan of @Obj. I personally feel @Component is more syntaxically close to Spring, which is what we're inspired one and what we should continue to follow.
- New map bukkit thing looks awesome
- Perhaps it'd be time to think about a new command system based on Brigadier? I can submit an issue with the proposed changes

LeeGodSRC committed:
Upgrade asm to 9.3
Custom user agent in MavenUtil.java

LeeGodSRC committed:
Fixed some bukkit thing in map module

LeeGodSRC commented:
The reason I decide to use `@Obj` instead of `@Component` is because how Component has conflict with adventure Component class, which cause awful coding experience when you are using both of them in a single class.
and the new Obj rework has made everything much more flexible. syntaxically it's moving away from spring but in structure it is much more closer

LeeGodSRC commented:
for Brigadier I don't really have an idea on how the structure is like, you can made an issue about it but I would like to accept a PR about rather than implementing it myself

terminalsin commented:
Also will player states be something that will be shipped with 0.6?

terminalsin reviewed (approved):
LGTM, make sure to read my suggestion above

LeeGodSRC commented:
> Also will player states be something that will be shipped with 0.6?

Currently I'm still thinking how player state are going behave under the current structure of fairy, I'll make an issue before working it tho!

terminalsin commented:
> > Also will player states be something that will be shipped with 0.6?
> 
> Currently I'm still thinking how player state are going behave under the current structure of fairy, I'll make an issue before working it tho!

There's a really simple approach to it. Each MCPlayer data contains the PlayerState. When the setPlayerState method is called, the user's inventory, experience, health, etc... are persisted to storage with the state's instance. At that point the loaded state then will load that same data (unless they override the shouldPersist() method in the PlayerState class) and then call the onLoad function of the state. At that point, the previous state closes all of its listeners whilst the new state registers them and binds any obj adequately.

This effectively allows users to create fast system between swapping from editor mode, mod mode, spectator mode, survival mode, player mode etc...

This furthermore allows for the creation of player traits and so and forth. It isn't too abstract, and isn't too robust. It allows enough wiggleroom to mess plenty with.

LeeGodSRC commented:
> > > Also will player states be something that will be shipped with 0.6?
> > 
> > 
> > Currently I'm still thinking how player state are going behave under the current structure of fairy, I'll make an issue before working it tho!
> 
> There's a really simple approach to it. Each MCPlayer data contains the PlayerState. When the setPlayerState method is called, the user's inventory, experience, health, etc... are persisted to storage with the state's instance. At that point the loaded state then will load that same data (unless they override the shouldPersist() method in the PlayerState class) and then call the onLoad function of the state. At that point, the previous state closes all of its listeners whilst the new state registers them and binds any obj adequately.
> 
> This effectively allows users to create fast system between swapping from editor mode, mod mode, spectator mode, survival mode, player mode etc...
> 
> This furthermore allows for the creation of player traits and so and forth. It isn't too abstract, and isn't too robust. It allows enough wiggleroom to mess plenty with.

Sounds right, can you write them down in #60 would be helpful!

LeeGodSRC committed:
Use URLClassLoaderAccess rather than old ExtendedClassLoader

LeeGodSRC committed:
rewrite gradle plugin: https://github.com/FairyProject/fairy-gradle-plugin

deleted publishSnapshot task and -SNAPSHOT versioning now requires manual naming

LeeGodSRC committed:
Changed GitHub workflow to fit current deploy structure
